src/UglyToad.PdfPig/Parser/FileStructure/CrossReferenceObjectOffsetValidator.cs

```diff
@@ -24,11 +24,11 @@ public static bool ValidateCrossReferenceOffsets(IInputBytes bytes, CrossReferen
                 return true;
             }
 
-            var builderOffsets = new Dictionary<IndirectReference, long>();
-            
             var bruteForceOffsets = BruteForceSearcher.GetObjectLocations(bytes);
             if (bruteForceOffsets.Count > 0)
             {
+                var builderOffsets = new Dictionary<IndirectReference, long>();
```

Comment:
Moved this inside the if since it wasn't referenced outside

Reply:
What about doing `var builderOffsets = new Dictionary<IndirectReference, long>(bruteForceOffsets.Count);` here too?

Reply:
Agreed & updated!